### PR TITLE
sys: sort out ztimer_xtimer_compat and ztimer64_xtimer_compat depes

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR splits from [#17365](https://github.com/RIOT-OS/RIOT/pull/17365) all modules depending on 64bit features of xtimer. It allows not using `64bit` if not needed, note that because of `newlib` this is actually used most of the time (a nice case to optimize @kfessel)

In cases where there was already a ztimer based alternative for xtimer, those get used instead.

Note that some might still be missed as covered in the build but some of these modules.

### Testing procedure

- Green murdock

### Issues/PRs references

Split from [#17365](https://github.com/RIOT-OS/RIOT/pull/17365)

### CODE REVIEW

File: sys/Makefile.dep [hunk 1]
```diff
@@ -324,6 +335,13 @@ ifneq (,$(filter sema_inv,$(USEMODULE)))
   USEMODULE += atomic_utils
 endif
 
+ifneq (,$(filter sema,$(USEMODULE)))
+  ifneq (,$(filter ztimer_xtimer_compat,$(USEMODULE)))
+    # sema_timed required 64bit
+    USEMODULE += ztimer64_xtimer_compat
+  endif
+endif
```
Comment (fjmolinas) [inline sys/Makefile.dep:343]: With #17115 this dependency can be moved to only `tests/sema` and `posix_semaphore`

**Comment by fjmolinas** [comment]
> because of newlib this is actually used most of the time (a nice case to optimize @kfessel)

I think we could make `gettimeofday` part of a pseudomodule `newlib_syscalls_gettimeofday` and `native_syscalls_gettimeofday`, there probably are not many users in the tree, but I would rather leave this for a follow-up, it does not make the current situation worse, and with this PR as it is I know #17365 passes, and therefore I can move forward with #17721.

The `sema` change I would like to address in this PR if `lwip` change gets in.

File: sys/Makefile.dep (hunk 1 above)
Comment (kfessel) [reply sys/Makefile.dep:343]: lets wait for it

**Comment by kfessel** [comment]
gettimeofday might also be an easy port to ztimer_msec but this is for another PR

**Comment by fjmolinas** [comment]
> gettimeofday might also be an easy port to ztimer_msec but this is for another PR

I think it requires usec though...

**Comment by kfessel** [comment]
> > gettimeofday might also be an easy port to ztimer_msec but this is for another PR
> 
> I think it requires usec though...

https://pubs.opengroup.org/onlinepubs/009604599/functions/gettimeofday.html

"The resolution of the system clock is unspecified." -> we can do to the resolution what ever we want

actually 0 should be start of unix epoch might be the bigger posix breaker (it was changed somewhen)

**Comment by fjmolinas** [comment]
Seems like its all green here @kfessel

**Review by kfessel — APPROVED** [review]
rollin' rollin' rollin'

**Pull request merged**